homeassistant/components/sensor/airvisual.py

```diff
@@ -296,13 +300,20 @@ def update(self):
             if self.city and self.state and self.country:
                 resp = self._client.city(self.city, self.state,
                                          self.country).get('data')
+                self.longitude, self.latitude = resp.get('location').get(
+                    'coordinates')
             else:
                 resp = self._client.nearest_city(self.latitude, self.longitude,
-                                                 self.radius).get('data')
+                                                 self._radius).get('data')
             _LOGGER.debug('New data retrieved: %s', resp)
+
+            self.city = resp.get('city')
+            self.state = resp.get('state')
+            self.country = resp.get('country')
             self.pollution_info = resp.get('current', {}).get('pollution', {})
         except exceptions.HTTPError as exc_info:
-            _LOGGER('Unable to retrieve data from the API')
-            _LOGGER.error("There is likely no data on this location")
+            _LOGGER.error(
+                "Unable to retrieve data on this location: %s, %s, %s",
+                self.city, self.state, self.country)
```

Comment:
In the case that there's an error the first time using the coordinates, this error message will just show None for the city, state and country.

Reply:
That's actually expected: if someone puts in coordinates that don't line up with a location that AirVisual has data on, there shouldn't be a corresponding city/state/country.

Nevertheless, I've made a small change to how that data is logged.